src/main/java/org/terasology/scenario/internal/systems/ComponentEvaluation/EvaluationSystem.java

```diff
@@ -107,4 +122,86 @@ public void onConditionalCheckEvent(ConditionalCheckEvent event, EntityRef entit
 
         event.setResult(block1.equals(block2));
     }
+
+    @ReceiveEvent
+    public void onEvaluateItemPrefabEvent(EvaluateItemPrefabEvent event, EntityRef entity, ConstItemPrefabComponent component) {
+        event.setResult(prefabManager.getPrefab(component.prefabURI));
+    }
+
+    @ReceiveEvent
+    public void onEvaluateCountItem(EvaluateIntEvent event, EntityRef entity, ItemCountComponent component) {
+        Map<String, EntityRef> args = entity.getComponent(ArgumentContainerComponent.class).arguments;
+
+        //TODO: Replaced once evaluation of player is done
+        TriggeringEntityComponent temp = event.getPassedEntity().getComponent(TriggeringEntityComponent.class);
+        InventoryComponent invent = temp.entity.getComponent(InventoryComponent.class);
+
+
+        EvaluateItemPrefabEvent evalPrefab = new EvaluateItemPrefabEvent(event.getPassedEntity());
+        args.get("item").send(evalPrefab);
+        Prefab prefab = evalPrefab.getResult();
+
+        int count = 0;
+        for (EntityRef e : invent.itemSlots) {
+            if (e.exists() && e.getParentPrefab().exists()) {
+                if (e.getParentPrefab().equals(prefab)) {
+                    count++;
+                }
+            }
+
+        }
+
+        event.setResult(count);
+    }
+
+    @ReceiveEvent //TODO:Fix this, entity doesn't actually have displayname
+    public void onEvaluatePlayerName(EvaluateStringEvent event, EntityRef entity, PlayerNameComponent component) {
+        Map<String, EntityRef> args = entity.getComponent(ArgumentContainerComponent.class).arguments;
+
+        //TODO: Replaced once evaluation of player is done
+        TriggeringEntityComponent temp = event.getPassedEntity().getComponent(TriggeringEntityComponent.class);
+        DisplayNameComponent name = temp.entity.getComponent(DisplayNameComponent.class);
+
+        event.setResult(name.name);
+    }
+
+    @ReceiveEvent
+    public void onEvaluateComparator(EvaluateComparatorEvent event, EntityRef entity, ConstComparatorComponent comp) {
+        event.setResult(comp.compare);
+    }
+
+    @ReceiveEvent
+    public void onEvaluateIntComparison(ConditionalCheckEvent event, EntityRef entity, IntComparisonConditionalComponent comp) {
+        Map<String, EntityRef> args = entity.getComponent(ArgumentContainerComponent.class).arguments;
+
+        EvaluateIntEvent evalInt1 = new EvaluateIntEvent(event.getPassedEntity());
+        args.get("int1").send(evalInt1);
+        int int1 = evalInt1.getResult();
+
+        EvaluateIntEvent evalInt2 = new EvaluateIntEvent(event.getPassedEntity());
+        args.get("int2").send(evalInt2);
+        int int2 = evalInt2.getResult();
+
+        EvaluateComparatorEvent evalCompare = new EvaluateComparatorEvent(event.getPassedEntity());
+        args.get("compare").send(evalCompare);
+
+
+        switch (evalCompare.getResult()) {
+            case EQUAL_TO:

```

Comment:
Enum could have a evsluate method that takes 2 args and is implemented for each constant